resources/components/base/Button.vue

```diff
@@ -0,0 +1,228 @@
+<template>
+	<button
+		class="wbmad-button"
+		v-bind:class="classObject"
+		v-bind:disabled="disabled"
+		v-on:click="$emit( 'click' )"
+	>
+		<icon
+			v-if="icon"
+			v-bind:icon="icon"
+			v-bind:invert="invert"
+		/>
+		<slot />
+	</button>
+</template>
+
+<script>
+var Icon = require( './Icon.vue' );
+
+module.exports = {
+	name: 'Button',
+
+	components: {
+		icon: Icon
+	},
+
+	props: {
+		disabled: {
+			type: Boolean,
+			default: false
+		},
+		framed: {
+			type: Boolean,
+			default: true
+		},
+		icon: {
+			type: String,
+			default: null
+		},
+		// In OOUI, flags are passed in as an array (or a string or an object)
+		// and are handled by a separate mixin. Passing them in individually is
+		// a bit more readable and intuitive, plus it makes the code in this
+		// component simpler.
+		progressive: {
+			type: Boolean,
+			default: false
+		},
+		destructive: {
+			type: Boolean,
+			default: false
+		},
+		primary: {
+			type: Boolean,
+			default: false
+		}
+	},
+
+	computed: {
+		classObject() {
```

Comment:
Let's give this a more descriptive name, like `builtInClasses` or `buttonClasses` or something.

Reply:
I like the idea of using `builtInClasses` across components